### ISSUE TITLE
Google Play Services Permissions

### ISSUE THREAD

**Issue by Dror-Bar** [faizalshap/react-native-otp-verify#6]
Hey there @faizalshap 
First of all, thanks a lot for this package. It works great and I very much appreciate it.
I recently implemented this package in my app and noticed something a bit strange. When attempting to start the listener, I'd receive this message:
![screen shot 2018-12-25 at 12 54 23](https://user-images.githubusercontent.com/34343333/50421643-51884d00-084a-11e9-88ba-18059950995c.png)
Then it will not register the SMS. Only if I give Google Play Services access to **everything** will it work properly.
My question is, why is that? It would make sense that the only Permission I need is SMS.
Looking at the code, the only google implementation that I see is `com.google.android.gms:play-services-auth`. Does this library require ALL of the above Permissions? are there any additional implementations in this package that I'm missing, or is there nothing to do about this?
Thanks.

**Comment by faizalshap** [comment]
Thanks for appreciating bud :)
I am not sure why you are getting this. Usually with this package we don't need any permissions at all (Not even SMS ;)) 
It does need google play services though.
Can you please tell me which device, android version you are using ?

**Comment by Dror-Bar** [comment]
Yes of course.
By the way, it didn't happen to me but to another person (I had all permissions enabled by default). However, I noticed the same behaviour on my device if I removed some permissions from the google play services (such as access to camera or microphone for example).
I have Android 7.0 Moto G4.

I have noticed other than `com.google.android.gms:play-services-auth` there are also mentions of `com.google.android.gms:play-services-tasks` and `com.google.android.gms:play-services-base` in the file react-native-otp-verify.iml, maybe they require all of the permissions. I assume these are necessary though.
If it is I'll go ahead and close this issue.
